src/utils.h

```diff
@@ -13,4 +14,13 @@
 
 #define DBG() fprintf(stderr, "%s:%d\n", __func__, __LINE__)
 
+#define UNUSED __attribute__((unused))
+
+#define container_of(e, type, field) \
+	((type *)(uintptr_t)((char *)(e)-offsetof(type, field)))
```

Comment:
I slightly prefer CONTAINER_OF for function-like macros

Reply:
I have fixed this.

Still i'd notice that it's well known macro people used to write in lowercase. No one would treat it as a function.